src/util.rs

```diff
@@ -19,7 +19,13 @@ pub fn from_hex_se<S>(x: &U256, serializer: S) -> Result<S::Ok, S::Error>
 where
     S: Serializer,
 {
-    serializer.serialize_u64(x.as_u64())
+    /* try to convert to an unsigned 128-bit integer, otherwise strip high bits */
```

Comment:
can we just discard rather than strip high bits? Orders with an amount too high are just going to cause errors further down the line (eg in executioner). may be easier to just drop them here and return an error